README.md

```diff
@@ -337,6 +337,54 @@ IoT Core Shadows are human-readable JSON documents structured into `desired` and
 
 For the sample application, one node generates data, updating the `desired` section. The other node listens for updates to the `desired` section for its next target. Whenever it changes its state, it updates the `reported` state. The `delta` field is not used. The result is that the shadow shows an update to the `desired` digit every four seconds, with the `reported` digit moving towards the `desired` digit, alternating between upwards and downwards.
 
+### Sending ROS2 Logs to AWS CloudWatch
+
+The general strategy here is that we will send MQTT messages to a AWS IoT Rule using Basic Ingest, which in turn will forward the MQTT messages to Amazon Cloudwatch.
+
+We need to create an IAM role for the AWS IoT Rule. The first step is to set the permissions JSON file for the role.
+
+```
+export ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+
+export CWL_ROLE_PERMISSIONS_TEMPLATE="$(pwd)/templates/cwl_role_permissions_template.json"
+export CWL_ROLE_PERMISSIONS_FILE="$(pwd)/iot_certs_and_config/cwl_role_permissions.json"
+
+cat $CWL_ROLE_PERMISSIONS_TEMPLATE > $CWL_ROLE_PERMISSIONS_FILE
+
+sed -i -e "s/REGION/$AWS_REGION/g" $CWL_ROLE_PERMISSIONS_FILE
+sed -i -e "s/ACCOUNT_ID/$ACCOUNT_ID/g" $CWL_ROLE_PERMISSIONS_FILE
+cat $CWL_ROLE_PERMISSIONS_FILE
+```
+
+Next, create the role with the permissions JSON file we just created, along with the trust policy.
+
+
+```
+aws iam create-role --role-name IoTCloudWatchRole --assume-role-policy-document file://./templates/cwl_role_trust.json
+
+aws iam put-role-policy --role-name IoTCloudwatchRole --policy-name allow-publish-to-cwl --policy-document file://./iot_certs_and_config/cwl_role_permissions.json
+```
+
+Get the role ARN:
+
+```
+export CWL_ROLE_ARN=$(aws iam get-role --role-name  IoTCloudWatchRole --query 'Role.Arn')
```

Comment:
Is this something returned by the create-role command?

Reply:
Yup! the previous command `aws iam create-role --role-name IoTCloudWatchRole...` creates the role, then this fetches the arn with the same

Reply:
Can we collapse this down to one line where the previous command captures the ARN instead of executing a separate second command? That will match the style used elsewhere

Reply:
Done